### PR TITLE
fix(Turborepo): Flag off daemon for package discovery

### PR DESCRIPTION
**Pull request by gsoltis**
### Description

Daemon package watching has some bugs that need to be ironed out. Flag it off for now.

### Testing Instructions

Disabled integration test that explicitly checks for the behavior that we are flagging off.

Closes TURBO-2265

### CODE REVIEW

File: crates/turborepo-lib/src/run/mod.rs [hunk 1]
```diff
@@ -289,16 +290,32 @@ impl Run {
...
+
+            #[cfg(not(feature = "daemon-package-discovery"))]
+            let builder = builder.with_package_discovery(
+                LocalPackageDiscoveryBuilder::new(
+                    self.base.repo_root.clone(),
+                    None,
+                    Some(root_package_json.clone()),
+                )
+                .build()?,
+            );
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/run/mod.rs:315]: I think this is unnecessary as this is what package discovery gets chosen by default if one isn't provided for `PackageGraphBuilder`
```suggestion
```

File: crates/turborepo-lib/src/run/mod.rs [hunk 2]
```diff
@@ -273,6 +273,7 @@ impl Run {
         };
 
         // if we are forcing the daemon, we don't want to fallback to local discovery
+        //#[cfg(feature = "daemon-package-discovery")]
         let (fallback, duration) = if let Some(true) = self.opts.run_opts.daemon {
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/run/mod.rs:277]: If we're constructing `fallback`, we should make sure to use it. The call `LocalPackageDiscoveryBuilder.build()` can result in a `yarn --version` or `pnpm --version` if the user hasn't specified a `packageManager` in `package.json`.

Comment (gsoltis) [reply crates/turborepo-lib/src/run/mod.rs:277]: Ah, ok. I was trying to decide how much I needed to cut. Seems like cut all of the fallback stuff. I'll see what I can do.

**Review by chris-olszewski — APPROVED** [review]
Some unused imports, but those can be fix-shipped or fixed in follow up PR.

**Comment by gsoltis** [comment]
Will fix imports in next PR

**Comment by gsoltis** [comment]
Fixing imports: https://github.com/vercel/turbo/pull/7295

**Pull request merged**